static/app/views/issueDetails/groupActivity.tsx

```diff
@@ -1,219 +1,156 @@
+  const mutators = useMutateActivity({
+    organization,
+    group,
+  });
+
+  const deleteOptions: MutateOptions<TData, TError, TVariables, TContext> =
+    useMemo(() => {
+      return {
+        onError: () => {
+          addErrorMessage(t('Failed to delete comment'));
+        },
+        onSuccess: () => {
+          addSuccessMessage(t('Comment removed'));
+        },
+      };
+    }, []);
+
+  const createOptions: MutateOptions<TData, TError, TVariables, TContext> =
+    useMemo(() => {
+      return {
+        onError: () => {
+          addErrorMessage(t('Unable to post comment'));
+        },
+        onSuccess: data => {
+          GroupStore.addActivity(group.id, data);
+          addSuccessMessage(t('Comment posted'));
+        },
+      };
+    }, [group.id]);
+
+  const updateOptions: MutateOptions<TData, TError, TVariables, TContext> =
+    useMemo(() => {
+      return {
+        onError: () => {
+          addErrorMessage(t('Unable to update comment'));
+        },
+        onSuccess: data => {
+          const d = data as GroupActivityNote;
+          GroupStore.updateActivity(group.id, data.id, {text: d.data.text});
+          addSuccessMessage(t('Comment updated'));
+        },
+      };
+    }, [group.id]);
+
+  const handleDelete = useCallback(
+    (item: GroupActivityType) => {
+      const restore = group.activity.find(activity => activity.id === item.id);
+      const index = GroupStore.removeActivity(group.id, item.id);
+
+      if (index === -1 || restore === undefined) {
+        addErrorMessage(t('Failed to delete comment'));
+        return;
+      }
+      mutators.handleDelete(
+        item.id,
+        group.activity.filter(a => a.id !== item.id),
+        deleteOptions
+      );
+    },
+    [deleteOptions, group.activity, mutators, group.id]
+  );
+
+  const handleCreate = useCallback(
+    (n: NoteType, _me: User) => {
+      mutators.handleCreate(n, group.activity, createOptions);
+    },
+    [createOptions, group.activity, mutators]
+  );
+
+  const handleUpdate = useCallback(
+    (item: GroupActivityType, n: NoteType) => {
+      mutators.handleUpdate(n, item.id, group.activity, updateOptions);
+    },
+    [updateOptions, group.activity, mutators]
+  );
+
+  return (
+    <Fragment>
+      {(reprocessingStatus === ReprocessingStatus.REPROCESSED_AND_HASNT_EVENT ||
+        reprocessingStatus === ReprocessingStatus.REPROCESSED_AND_HAS_EVENT) && (
+        <ReprocessedBox
+          reprocessActivity={mostRecentActivity as GroupActivityReprocess}
+          groupCount={groupCount}
+          orgSlug={organization.slug}
+          groupId={groupId}
+        />
+      )}
+
+      <Layout.Body>
+        <Layout.Main>
+          <ActivitySection
+            group={group}
+            mutators={{
+              onDelete: handleDelete,
+              onCreate: handleCreate,
+              onUpdate: handleUpdate,
+            }}
```

Comment:
can break these into their own props since we don't get anything by grouping them into mutators